body/stretch_body/dynamixel_hello_XL430.py

```diff
@@ -453,6 +453,29 @@ def move_to_vel(self,v_des,a_des=None):
                 if self.bubble_up_comm_exception:
                     raise DynamixelCommError
 
+    def set_velocity(self,v_des,a_des=None):
+        nretry = 2
+        if not self.hw_valid:
+            return
+        if self.params['req_calibration'] and not self.is_calibrated:
+            self.logger.warning('Dynamixel not calibrated: %s' % self.name)
+            print('Dynamixel not calibrated:', self.name)
+            return
+        success = False
+        for i in range(nretry):
+            try:
+                self.motor.disable_torque()
+                self.motor.enable_vel()
+                self.motor.enable_torque()
```

Comment:
The [DynamixelXL430.enable_vel()]() is an EEPROM table write operation, which delays the set_velocity() method execution when called within a control loop which makes the motion jittery and inaccurate. 
Verified with this behaviour with  [test_dxl_vel.py](https://github.com/hello-robot/stretch_body/blob/bugfix/dxl_vel/body/test/test_dxl_vel.py) which tests velocity control accuracy using higher-level [Dynamixel.HelloXL430.set_velocity()](https://github.com/hello-robot/stretch_body/blob/bugfix/dxl_vel/body/stretch_body/dynamixel_hello_XL430.py#L456) compared to using the lower-level [DynamixelHelloXL430.set_vel()](https://github.com/hello-robot/stretch_body/blob/bugfix/dxl_vel/body/stretch_body/dynamixel_XL430.py#L582) within a control loop.
```suggestion
```

Reply:
@hello-binit Could you confirm if the set_velocity() is not intended not to be used in a control loop?

Reply:
Set_velocity most likely would be used in a >10hz control loop, so you're correct in taking the EEPROM calls out of the method. Instead of creating a separate method to change into velocity mode, I think the set_velocity method should check the operating mode and change into velocity mode only when needed. This would make it so only motor.get_operating_mode() gets called at the control rate. If calling get_operating_mode() still causes jitter, I think we can cache the current control mode in the DynamixelXL430Hello class. Wherever the operating mode is changed, a variable is also updated. Then move_to/by can check the variable to see if the joint is in position/multiturn mode before sending a command, and set_velocity can check if the joint is in velocity mode before sending a command.

Reply:
The [`motor.get_operating_mode()`](https://github.com/hello-robot/stretch_body/blob/bugfix/dxl_vel/body/stretch_body/dynamixel_hello_XL430.py#L465) doesn't seem to affect the motion when called every time and performs same as accessing the lower level method being an EEPROM read operation. The velocity mode is enabled [here](https://github.com/hello-robot/stretch_body/blob/bugfix/dxl_vel/body/stretch_body/dynamixel_hello_XL430.py#L466) only after checking the operating mode now. Thanks for the suggestions.